core/src/main/java/org/apache/accumulo/core/data/Mutation.java

```diff
@@ -66,18 +68,44 @@
    */
   static final int VALUE_SIZE_COPY_CUTOFF = 1 << 15;
 
+  /**
+   * Maximum size of a mutation (2GB).
+   */
+  static final long MAX_MUTATION_SIZE = 2l * (1 << 30);
```

Comment:
Minor nit: prefer uppercase L for long literals. It's easier to read, especially for programmers with poor eyesight.

Also, `1L << 31` is simpler.